Sources/Brave/Frontend/Browser/Toolbars/BottomToolbar/Menu/HistoryViewController.swift

```diff
@@ -199,6 +202,11 @@ class HistoryViewController: SiteTableViewController, ToolbarUrlActionsProtocol
             // Clearing History should clear Recently Closed
             RecentlyClosed.removeAll()
           }
+          
+          // Asking Sync Engine To Remove Visits
+          for historyItems in allHistoryItems {
+            self.historyAPI.removeHistory(historyItems)
```

Comment:
Hmm, when removing `ALL` history, is there no API in sync to ask it to remove all instead of the specific history above? Maybe we can export an API that can just remove all the history without the need to get the `fetchedObjects`?

Maybe not for now this PR though, but might be worth it in the future.

Reply:
The problem is this new BrowsingHistoryService doesnt have it. I checked ofcourse and normal History service method doesnt trigger Sync delete directives.

But later we can adjust add methods also using BrowsingHistoryService but I dont see any hurry to do it for bnow like you said.